crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -23,50 +23,54 @@ use crate::config::{NameServerConfigGroup, ResolverConfig, ResolverOpts, ServerO
 use crate::error::{ResolveError, ResolveErrorKind};
 #[cfg(feature = "mdns")]
 use crate::name_server;
-use crate::name_server::{ConnectionProvider, NameServer};
+use crate::name_server::name_server::{AbstractNameServer, CreateConnection};
 #[cfg(test)]
 #[cfg(feature = "tokio-runtime")]
-use crate::name_server::{TokioConnection, TokioConnectionProvider, TokioHandle};
+use crate::name_server::TokioRuntimeProvider;
+use crate::name_server::{GenericConnection, RuntimeProvider};
 
-/// A pool of NameServers
-///
-/// This is not expected to be used directly, see [crate::AsyncResolver].
+/// Abstract interface for mocking purpose
 #[derive(Clone)]
-pub struct NameServerPool<
-    C: DnsHandle<Error = ResolveError> + Send + Sync + 'static,
-    P: ConnectionProvider<Conn = C> + Send + 'static,
+pub struct AbstractNameServerPool<
```

Comment:
Here, too, I think we should switch the naming around, to leave this called `NameServerPool`. At least, you mentioned `GenericConnection` was mainly used for testing, right?

Reply:
Not accurate. The `GenericConnection` is the actual implementation. My previous question is actually "Can I remove the `GenericConnection` by moving its code to proper places". But I cannot get rid of it at last.